### PR TITLE
Create connection factory

### PR DESCRIPTION
**Pull request by webrouse**
Changes:
- Create `MysqlDbConnectionFactory`, moved code from `MySQL.php`.
- Factory is used in tests to setup SSL MySQL server.

### CODE REVIEW

File: src/Keboola/DbExtractor/Extractor/MySQLDbConnectionFactory.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                    $pdo->exec('SET NAMES utf8mb4;');
+                } catch (PDOException $exception) {
+                    $logger->info('Falling back to "utf8" charset');
+                    $pdo->exec('SET NAMES utf8;');
+                }
+
+                // Set isolation level
+                if ($dbConfig->hasTransactionIsolationLevel()) {
+                    $pdo->query(
+                        sprintf('SET SESSION TRANSACTION ISOLATION LEVEL %s', $dbConfig->getTransactionIsolationLevel())
```
Comment (webrouse) [inline src/Keboola/DbExtractor/Extractor/MySQLDbConnectionFactory.php:89]: Tato cast kodu bola povodne o uroven vyssie a nie v tomto callbacku, ... no ak by nastal vypadok spojenia a reconnect, tak by sa to uz nenastavilo. Preto som to presunul.

File: src/Keboola/DbExtractor/Extractor/MySQL.php [hunk 2]
```diff
@@ -23,13 +23,6 @@
 
 class MySQL extends BaseExtractor
 {
-    // Some SSL keys who worked in Debian Stretch (OpenSSL 1.1.0) stopped working in Debian Buster (OpenSSL 1.1.1).
```
Comment (webrouse) [inline src/Keboola/DbExtractor/Extractor/MySQL.php:26]: Presunute do connection factory, vysvetlene dalej.

File: tests/functional/DatadirTest.php [hunk 3]
```diff
@@ -55,7 +55,8 @@ protected function setUp(): void
         $this->testProjectDir = $this->getTestFileDir() . '/' . $this->dataName();
         $this->testTempDir = $this->temp->getTmpFolder();
 
-        $this->connection = PdoTestConnection::createConnection();
+        $isSsl = strpos((string) $this->dataName(), 'ssl-') === 0;
+        $this->connection = PdoTestConnection::createConnection($isSsl);
```
Comment (webrouse) [inline tests/functional/DatadirTest.php:59]: Chcem vytvorit `mysql-ssl` server pre SSL testy, ktory ma vynutene SSL spojenie.
No ak to spravim, tak mi neprejdu testy, ... pretoze testy pouzivaju nie-SSL spojenie.

Preto tato zmena -> ak testy zacina `ssl-`, tak sa databaza nastavi cez SSL spojenie.
A preto som musel vytvorit connection factory, aby som nemal duplicitny kod.

File: tests/functional/PdoTestConnection.php [hunk 4]
```diff
@@ -7,71 +7,49 @@
...
-
-        $dsn = sprintf(
-            'mysql:host=%s;port=%s;charset=utf8',
-            $databaseConfig->getHost(),
-            $port
-        );
-
-        if ($databaseConfig->hasDatabase()) {
-            $dsn = sprintf(
-                'mysql:host=%s;port=%s;dbname=%s;charset=utf8',
```
Comment (webrouse) [inline tests/functional/PdoTestConnection.php:55]: Povodne vytvaranie spojenia v testoch -> bez SSL.

**Review by ondrajodas — APPROVED** [review]
ok

**Pull request merged**